core/arch/arm/kernel/kern.ld.S

```diff
@@ -394,6 +394,7 @@ SECTIONS
 	__init_size = __data_end - CFG_TEE_LOAD_ADDR;
 	__init_mem_usage = __end - CFG_TEE_LOAD_ADDR;
 #endif
+	ASSERT(. <= (CFG_TEE_RAM_START + CFG_TEE_RAM_VA_SIZE), "the CFG_TEE_RAM_VA_SIZE is not enough!")
```

Comment:
@etienne-lms the next line won't print any error if CFG_TEE_RAM_VA_SIZE is too small, apparently (try it with CFG_TEE_RAM_VA_SIZE == 0 for instance!). The commit message should have stated that.
I must admit I don't understand why, because you're not supposed to be able to move the location counter backwards. So I'd like to understand what's going on before adding an assertion.